### PR TITLE
Fix documentation of template instantiations

### PR DESCRIPTION
**Pull request by dangars**
The instantiations for long double and std::complex<long double> were removed in commit 352537aca

I modified the documentation. Let me know if you are happy with the new text!

### CODE REVIEW

File: cmake/scripts/expand_instantiations.cc [hunk 1]
```diff
@@ -19,11 +19,10 @@
...
+// REAL_SCALARS    := { double; float}
 // COMPLEX_SCALARS := { std::complex<double>;
-//                      std::complex<float>;
-//                      std::complex<long double> }
-// VECTORS := { Vector<double>; Vector<float>; Vector<long double> }
+//                      std::complex<float>}
+// VECTORS := { Vector<double>; Vector<float>}
 // --------------------
 //
 // The input file is typically located in share/deal.II/template-arguments in
```
Comment (drwells) [inline cmake/scripts/expand_instantiations.cc]: While you are here, would you mind also fixing this path?

Comment (drwells) [reply cmake/scripts/expand_instantiations.cc]: This file has long since been moved to `cmake/config/template-arguments.in`.

Comment (dangars) [reply cmake/scripts/expand_instantiations.cc]: Done!

Comment (drwells) [reply cmake/scripts/expand_instantiations.cc]: Sorry, I put my comment on the wrong line before: the previous comment was correct (the file really is at `share/deal.II/template-arguments` in the build directory) but the next clause should be `and it is built from cmake/config/template-arguments.in`.

**Comment by dangars** [comment]
Hi @drwells Thanks for the correction. Is it ok now?

**Comment by drwells** [comment]
Yup!

**Pull request merged**